firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReport.java

```diff
@@ -0,0 +1,854 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.model;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Application.Organization;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event.Application;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event.Application.Execution.Thread.Frame;
+import com.google.firebase.encoders.annotations.Encodable;
+import com.google.firebase.encoders.annotations.Encodable.Field;
+import com.google.firebase.encoders.annotations.Encodable.Ignore;
+import java.nio.charset.Charset;
+import java.util.List;
+
+@Encodable
+@AutoValue
+public abstract class CrashlyticsReport {
```

Comment:
This top-level class really need a javadoc comment explaining how it works. It looks like the rest of the classes are mostly AutoValue representations of the report model, so maybe it's not necessary to have comments for all of them, but definitely at least one at the top.

Reply:
Agreed. Will fix.